### ISSUE TITLE
Question: Is it possible to make my game interact with the index?

### ISSUE THREAD

**Issue by rafael-lua** [TannerRogalsky/love.js#69]
I couln't find a solution (if there is any) with my limited skills, so I am here to see if someone could help.

It is possible to make my game interact with scripts in the index page? So I could use things like https://docs.kongregate.com/docs/javascript-api

Everything works fine with the build. But I would really want to be able to integrate the kong api or even to execute some javascript code and retrieve the result to my game. Love2d is a engine I really like and I don't want to change it and start all over in another place.

Is there any way to my game code communicate with the outside page?

**Comment by MrcSnm** [comment]
Okay, you got my day today, I spent my day trying this and found out how to do it, basically, what it does is wrap the console log function and then use it to evaluate, it is really easy to use, and I'm also doing it with kongregate

https://github.com/MrcSnm/Love.js-Api-Player

This is an example of how to use it

```lua
if(key == "p") then
        callJS("kongregate.stats.submit('Score', 1000);")
    end
```

![apiTest](https://user-images.githubusercontent.com/10136262/84332331-0f002080-ab63-11ea-8193-a5f9667b246d.png)

**Comment by rafael-lua** [comment]
> Be sure to watch it, I'll post one PR later for the focus problem when uploading your game onto kongregate

I don't believe, someone actually did it! This is so amazing, thank you so much!!! I can keep working with love and Lua even for web, I'm truly grateful. 

I will make sure to keep watching it!

**Comment by MrcSnm** [comment]
> > Be sure to watch it, I'll post one PR later for the focus problem when uploading your game onto kongregate
> 
> I don't believe, someone actually did it! This is so amazing, thank you so much!!! I can keep working with love and Lua even for web, I'm truly grateful.
> 
> I will make sure to keep watching it!

You're welcome, I'm doing a kongregate game myself so I needed to implement it anyway
The new PR is ready, added some easier compiling function and it already comes with the <script> tags for the api call

**Comment by MrcSnm** [comment]
Just passing by to say that I made massive changes to **[Love JS API Player](https://github.com/MrcSnm/Love.js-Api-Player)**
Now we're able to just not send requests, but to retrieve its value, the Lib is really simple to use and you can design it for being asynchronous or synchronous
Although I didn't include the "Percentage complete" feature, it is quite easy to understand how it works

**Comment by MrcSnm** [comment]
Just passing by to say that I did the final update on [Love JS API Player](https://github.com/MrcSnm/Love.js-Api-Player)

Now it is possible to retrieve data from promises and execute javascript code blocks:
```lua
JS.newPromiseRequest(JS.stringFunc(
    [[
        var xhttp = new XMLHttpRequest();
        xhttp.onreadystatechange = function()
        {    
            if (this.readyState == 4 && this.status == 200)
                _$_(this.responseText);
        };
        xhttp.open("GET", "https://example.com", true);
        xhttp.send();
    ]]
```
